app/org/sagebionetworks/bridge/stormpath/StormpathDirectoryDao.java

```diff
@@ -224,66 +189,49 @@ private void adjustPasswordPolicies(Study study, Directory directory) {
-            ObjectNode directoryNode = StormpathUtils.getJSON(client, directory.getHref());
-            String accountCreationUrl = directoryNode.get("accountCreationPolicy").get("href").asText();
-            
-            // Get account policy as JSON, update to our standard configuration
-            ObjectNode accountPolicyNode = StormpathUtils.getJSON(client, accountCreationUrl);
-            String verificationEmailTemplatesUrl = accountPolicyNode.get("verificationEmailTemplates").get("href").asText();
-            accountPolicyNode.put("verificationEmailStatus", "ENABLED");
-            accountPolicyNode.put("verificationSuccessEmailStatus", "DISABLED");
-            accountPolicyNode.put("welcomeEmailStatus", "DISABLED");
-            // save the account policy
-            StormpathUtils.postJSON(client, accountCreationUrl, accountPolicyNode);
-            
-            // Get the verify email template
-            ObjectNode templateNode = StormpathUtils.getJSON(client, verificationEmailTemplatesUrl);
-            String templateUrl = templateNode.get("items").get(0).get("href").asText();
-            // Update this template with study-specific information
-            ObjectNode template = StormpathUtils.getJSON(client, templateUrl);
-            if (study.getSponsorName() != null) {
-                template.put("fromName", study.getSponsorName());    
-            } else {
-                template.put("fromName", study.getName());
-            }
-            template.put("fromEmailAddress", study.getSupportEmail());
-            template.put("subject", partiallyResolveTemplate(ve.getSubject(), study));
-            template.put("mimeType", ve.getMimeType() == MimeType.HTML ? "text/html" : "text/plain");
-            String body = partiallyResolveTemplate(ve.getBody(), study);
-            template.put("textBody", body);
-            template.put("htmlBody", body);
-            String link = String.format("%s/mobile/verifyEmail.html?study=%s", config.getBaseURL(), study.getIdentifier());
-            ((ObjectNode)template.get("defaultModel")).put("linkBaseUrl", link);
-            // save the verify email template
-            StormpathUtils.postJSON(client, templateUrl, template);
-        } catch(Throwable throwable) {
-            throw new BridgeServiceException(throwable);
+        AccountCreationPolicy policy = directory.getAccountCreationPolicy();
+        
+        // According to the documentation, there is only one...
+        ModeledEmailTemplate template = policy.getAccountVerificationEmailTemplates().iterator().next();
+        updateTemplate(study, template, study.getVerifyEmailTemplate(), "verifyEmail");
+        
+        policy.setVerificationEmailStatus(EmailStatus.ENABLED);
+        policy.setWelcomeEmailStatus(EmailStatus.DISABLED);
+        policy.save();
+    }
+    
+    private void updateTemplate(Study study, ModeledEmailTemplate stormpathTemplate, EmailTemplate template, String pageName) {
```

Comment:
A lot of these methods don't depend on object members. Would it be worth making them static to give them the most restrictive semantics?